prom.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+
+	log "github.com/Sirupsen/logrus"
+	prom "github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+type promMessage struct {
+	registry *prom.Registry
+}
+
+type promIO struct {
+	scrapeSignalChan chan bool
+	messageChan      chan promMessage
+}
+
+func resetRegistry(registry *prom.Registry) {
+	prom.DefaultRegisterer = registry
```

Comment:
thanks for finding this one, I had being using it before I learned about a way to expose non-global registry.